coremltools/converters/nnssa/coreml/graph_pass/mlmodel_passes.py

```diff
@@ -244,27 +245,49 @@ def remove_redundant_transposes(spec):
     that compose to the identity.
     """
 
+    def _get_input_output_hashmap(nn_layers):
```

Comment:
This method does not handle the `loop` and `branch` layers that can hold nn_spec inside them. 
So to cover all layers you would have to make it a recursive function and recurse with the layer type. 
For something similar, look at this method that prints the description of of the NN spec: https://github.com/apple/coremltools/blob/437fa2b534169a362ac6b98750ac19a616d097de/coremltools/models/neural_network/spec_inspection_utils.py#L183

Reply:
(or you could ignore handling the loop and branch layers for now, it will make things complicated for the rest of the code, I guess that can be handled in a later PR)

Reply:
I think we can handle this issue in the future PR.